be/src/olap/rowset/segment_v2/segment.cpp

```diff
@@ -522,24 +522,17 @@ Status Segment::new_column_iterator_with_path(const TabletColumn& tablet_column,
     auto sparse_node = tablet_column.has_path_info()
                                ? _sparse_column_tree.find_exact(*tablet_column.path_info_ptr())
                                : nullptr;
-    if (opt != nullptr && opt->io_ctx.reader_type == ReaderType::READER_ALTER_TABLE) {
```

Comment:
Why READER_ALTER_TABLE does not need to be prcessed specially as before?

Reply:
it's not special now